src/execution/physical_plan/plan_copy_to_file.cpp

```diff
@@ -6,9 +6,10 @@ namespace duckdb {
 
 unique_ptr<PhysicalOperator> PhysicalPlanGenerator::CreatePlan(LogicalCopyToFile &op) {
 	auto plan = CreatePlan(*op.children[0]);
+	op.file_path += ".tmp";
 	// COPY from select statement to file
```

Comment:
Can we add an option for this behavior (e.g. the original `OVERWRITE`)? There are some reasons why you might not want this behavior (e.g. potential to leave an extra file behind, have to fit both files on disk). Perhaps defaulting to false is safest, although I can see the argument for defaulting to true as long as we correctly detect e.g. the pipe scenario.

Reply:
I'm not so sure if this flag makes that much sense. 

I would say that cleaning up  .tmp files of failed queries should be enough since this will always be garbage from the query perspective anyway.

The two files on disk would only exist during query execution, which I think should be perfectly fine to all but 0.001% of scenarios, no? Otherwise, the user should do some disk cleanup.

Reply:
The file will be left behind when the system crashes and can't clean up. Perhaps not enough of a problem, but definitely a downside. Since we need to be able to turn this functionality off anyway (for when writing to a pipe fs or to a httpfs directly) I think having this flag there makes sense. But I agree that perhaps it should default to true.

Reply:
well, a system crash sounds more like a reason to have .tmp file than an overwrite.
I see the point for turning it off though, having the default to true seems the way go to.

One last thing, overwrite might be a bad name, if you want to expose that to a user.

Reply:
That depends if the user cares about preserving the original file, which won’t always be the case. Perhaps “use_temp_file” is a better name?